### ISSUE TITLE
Issue remembering previously accepted OAuth2 scopes

### ISSUE THREAD

**Issue by CHTJonas** [camdram/camdram#622]
Something is wrong with external OAuth login for certain user(s). Unfortunately the error messages aren't very forgiving...

It also seems to work fine for me, but not for the user in question below (won't mention their name here for data protection but see Sentry). I'll try and ask a few other people to test this but to help diagnose it might help if people could try navigating to https://roombooking-dev.adctheatre.com and logging in to see if you experience the same issue?

Sentry issue: [CAMDRAM-WEB-8B](https://sentry.io/organizations/camdram-webteam/issues/840240214/?project=1273126&referrer=github_plugin)

```
OAuth2\OAuth2ServerException: invalid_request
  File "web/app.php", line 28
    $response = $kernel->handle($request);
...
(8 additional frame(s) were not displayed)
```

**Comment by CHTJonas** [comment]
Seems to be line 1115 of `vendor/friendsofsymfony/oauth2-php/lib/OAuth2.php` which is the issue:
```php
throw new OAuth2ServerException(Response::HTTP_BAD_REQUEST, self::ERROR_INVALID_REQUEST, "No client id supplied");
```
But the POST'ed request body clearly shows the `client_id` is present?

**Comment by CHTJonas** [comment]
I can login without issue via OAuth2 on the [CUADC Wiki](https://wiki.cuadc.org/wiki/Main_Page) so I don't think it's a client application problem.

**Comment by CHTJonas** [comment]
I was playing around and ran the following query to purge the stored user grants and I'm now experiencing the same issue:
```sql
DELETE FROM acts_api_authorizations;
```

**Comment by CHTJonas** [comment]
After a few hours of messing around, I believe I have narrowed down the problem. The first time a user attempts to authenticate using Camdram to an external application using OAuth2, the "The app XXX would like to access your profile on Camdram" message appears on Camdram. At this point the "Allow" and "Cancel" buttons work as expected. However upon clicking said "Allow" button, the required scopes are not stored in the `acts_api_authorizations` table - instead the `scopes` field is set to be a blank string.

When attempting to log in to the app in question again at a later date, the same screen appears again. Upon clicking either the "Allow" or "Cancel" buttons an exception is thrown. If I manually set the `scopes` field to what I expect it to be `UPDATE acts_api_authorizations SET scopes = 'user_shows,user_orgs,user_email' WHERE id=50` then everything works as expected.
